ipatests/prci_definitions/prci_checker.py

```diff
@@ -0,0 +1,254 @@
+        )
+        sys.exit(1)
+    except yaml.YAMLError as e:
+        print(e)
+        print_error(f"Error loading YAML definition file {file}")
+        sys.exit(1)
+
+    # Get spec for file to be analyzed
+    current_spec = jobs_spec.get(filename)
+    if current_spec is None:
+        print_warning(
+            f"'{filename}' file is not defined in the PRCI "
+            "specification file and "
+            "could not be analyzed."
+        )
+        return True
+
+    return check_jobs(filename, def_suite, current_spec)
+
+
+def parse_arguments(description):
+    """Parse and return arguments if specified"""
+    parser = argparse.ArgumentParser(
+        description=description, formatter_class=RawTextHelpFormatter
+    )
+    parser.add_argument(
+        "-f", "--file", help="Specify YAML definition file to be analyzed"
+    )
+    parser.add_argument(
+        "-s",
+        "--spec",
+        default=JOBS_SPEC_YAML,
+        help="Specify custom path for specification file",
+    )
+    return parser.parse_args()
+
+
+def main():
+    """
+    Checker script for prci definition files.\n
+    This script checks whether jobs in a prci definition file have the correct
+    naming format, requirements, and arguments, which are defined in the
+    specification file.
+
+    Examples of the usage for the tool:\n
+    # Check all yaml definition files in current dir\n
+    python3 prci_checker.py\n
+    # Check only specified file\n
+    python3 prci_checker.py -f gating.yaml\n
+    # Check with custom path for spec file\n
+    python3 prci_checker.py -s ../../alternative_spec.yaml
+    """
+    args = parse_arguments(main.__doc__)
+
+    print("BEGINNING PRCI JOB DEFINITIONS CHECKS")
+
+    # Load file with jobs specification format
+    try:
+        spec_root = load_yaml(args.spec)
+    except FileNotFoundError as e:
+        print(e)
+        print_error(f"Jobs specification file '{args.spec}' not found.")
+        sys.exit(1)
+    except yaml.YAMLError as e:
+        print(e)
+        print_error(f"Error loading YAML specification file '{args.spec}'")
+        sys.exit(1)
+
+    jobs_spec = spec_root.get("prci_job_spec")
+    if not jobs_spec:
+        print_error(
+            f"Specification definition not found in spec file '{args.spec}'\n"
+            "  Key 'prci_job_spec' is not present."
+        )
+        sys.exit(1)
+
+    if args.file:
+        result = process_file(args.file, jobs_spec)
+    else:
+        # Get all yaml files in current dir, except IGNORE_FILES
+        defs_files = glob.glob("*.yaml")
+        defs_files = set(defs_files) - IGNORE_FILES
+        if not defs_files:
+            print_warning(
+                "No yaml job definition files found to analyze "
+                "in current directory."
+            )
+            return
+
+        result = True
+        for def_file in defs_files:
+            if not process_file(def_file, jobs_spec):
+                result = False
```

Comment:
Thanks for the suggestion, I didn't know about that method, it's very convenient.
I have kept the regular for loop because it's now more complex after adding the job counting.